Source/Core/Core/PowerPC/Jit64/FPURegCache.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/PowerPC/Jit64/FPURegCache.h"
+
+#include "Core/PowerPC/Jit64/Jit.h"
+#include "Core/PowerPC/Jit64Common/Jit64Base.h"
+#include "Core/PowerPC/Jit64Common/Jit64PowerPCState.h"
+
+using namespace Gen;
+
+FPURegCache::FPURegCache(Jit64& jit) : RegCache{jit}
+{
+}
+
+void FPURegCache::StoreRegister(size_t preg, const OpArg& new_loc)
+{
+  m_emitter->MOVAPD(new_loc, m_regs[preg].location.GetSimpleReg());
+}
+
+void FPURegCache::LoadRegister(size_t preg, X64Reg new_loc)
+{
+  m_emitter->MOVAPD(new_loc, m_regs[preg].location);
+}
+
+const X64Reg* FPURegCache::GetAllocationOrder(size_t* count)
```

Comment:
Is there a reason this function is like this?  I guess it's down to personal opinion but I don't usually like to return a second value from the function unless it's super necessary.  I'd suggest splitting up the count into another function (GetAllocationOrderSize?).  I'm not a performance guy but you might be able to even inline it (depending on how much this will be used)?

You'd need to move from a function static variable for "allocation_order" to an anonymous namespace or something so that both functions could call it.  Or use a static member..